lib/cartopy/crs.py

```diff
@@ -1356,6 +1392,30 @@ def __init__(self, central_latitude=0.0, central_longitude=0.0,
             self._boundary = sgeom.LinearRing(coords.T)
         self._threshold = np.diff(self._x_limits)[0] * 1e-3
 
+    @classmethod
+    def from_proj4(cls, proj4_dict, **kwargs):
+        from cartopy._proj4 import _globe_from_proj4
+        globe = _globe_from_proj4(proj4_dict)
+        p_kwargs = {'globe': globe}
+
+        if 'lon_0' in proj4_dict:
+            p_kwargs['central_longitude'] = float(proj4_dict['lon_0'])
+        if 'lat_0' in proj4_dict and cls is Stereographic:
+            # forced by North and South specific classes
+            p_kwargs['central_latitude'] = float(proj4_dict['lat_0'])
+        if 'lat_ts' in proj4_dict:
+            p_kwargs['true_scale_latitude'] = float(proj4_dict['lat_ts'])
+        if 'k_0' in proj4_dict:
+            p_kwargs['scale_factor'] = float(proj4_dict['k_0'])
+        if 'x_0' in proj4_dict:
+            p_kwargs['false_easting'] = float(proj4_dict['x_0'])
+        if 'y_0' in proj4_dict:
+            p_kwargs['false_northing'] = float(proj4_dict['y_0'])
```

Comment:
There is repeated handling of things like ``false_northing`` happening. Can we factorise this?

Reply:
I thought about that, but for this initial commit I didn't do it because it felt like I would be trying to hard to reduce duplicate code and I wasn't sure if all CRS classes supported the false easting and northing.

Reply:
Not all projections do support false_easting / false_northing, but those that do all map to ```y_0```. In some respects, if the ```proj4_terms``` includes this, then whether or not the projection supports it I'd be tempted to try passing the keyword on to the ```__init__``` anyway. At least then the user gets a message about ```false_northing``` not being a valid keyword.

What would be missing from this approach is the user knowing that ```y_0``` was the term that mapped to ```false_northing```. Perhaps we could make use of ```inspect``` to determine ahead-of-time what the ```__init__``` supports?